x/auth/ante/ante_test.go

```diff
@@ -23,7 +27,7 @@ import (
 
 // Test that simulate transaction accurately estimates gas cost
 func (suite *AnteTestSuite) TestSimulateGasCost() {
-	suite.SetupTest(true) // reset
```

Comment:
why do we change this in all tests?

Reply:
we've run a diff check on covered lines and they match either if setupTest is run with a true or fals isCheckTx param.

in reality what 'true' does is that it correctly inits genesis. which we require because if we do not the supply will be nil which will cause a panic when we mint coins.

Reply:
Hmm, Why minting new coins requires a genesis?

Reply:
Because supply is NIL, this will cause a panic from the bank keeper if tries to get a nil supply. I wanted to address this  but the consensus with aaron was that eventually when supply by denom PR is implemented this would ahve been automatically fixed.